### PR TITLE
Always use host netns

### LINKED ISSUE #1589: fastdp broke multiweave
```
$ NUM_WEAVES=3 bin/multiweave launch
65fd3e8d118f35599fa5b033a88ab6f374fcba73c6921f9485d4dfabc860387c
1ac7cca04cdc1ee5713a741cb5177f49f31c738f13f77e50e6536ea087ba24ea
ff804a0d2ddf22d5b2cf42da5a4d884293510538f5cb8fb098001bf82eed938e
connecting weave3 to false
weave3 container is not running.
$ docker logs weave3
...
FATA: 2015/10/25 15:55:39.786639 listen udp4 :6783: bind: address already in use
```

### PR DESCRIPTION
**Pull request by awh**
Fixes #1589,#1746.

~~This PR currently depends on an as yet unmerged change in mesh (weaveworks/mesh#5); that should be merged first and this PR rebased accordingly.~~

### CODE REVIEW

File: weave [hunk 1]
```diff
@@ -829,16 +832,12 @@ setup_router_iface_fastdp() {
 }
 
 setup_router_iface_bridge() {
-    if ! netnsenter ip link show eth0 >/dev/null ; then
-        echo "Perhaps you are running the docker daemon with container networking disabled (-b=none)." >&2
+    if ! ip link add name $BRIDGE_IFNAME mtu $MTU type veth peer name $PCAP_IFNAME mtu $MTU ||
+       ! ip link set $BRIDGE_IFNAME up ||
+       ! add_iface_bridge $BRIDGE_IFNAME ||
+       ! ip link set $PCAP_IFNAME up ; then
```
Comment (rade) [inline weave]: This all needs to be a) idempotent, and b) clean up on failures.

```
$ weave reset
$ WEAVE_NO_FASTDP=1 weave launch
$ weave stop
$ WEAVE_NO_FASTDP=1 weave launch
RTNETLINK answers: File exists
$ echo $?
1
```

**Comment by awh** [comment]
Addressed comments and updated mesh dependency.

**Comment by rade** [comment]
I think `connect_container_to_bridge` can be simplified now that it only gets called for application containers and hence always operates on a proper netns. In particular the interface 'down'ing (and elaborate explanatory comment) should no longer be necessary.

**Comment by rade** [comment]
The old multiweave lets one launch additional peers, and also stop just some of the peers, by setting `FIRST_WEAVE`, e.g.

```
NUM_WEAVES=10 bin/multiweave launch
FIRST_WEAVE=11 NUM_WEAVES=10 bin/multiweave launch
FIRST_WEAVE=11 NUM_WEAVES=10 bin/multiweave stop
```

This is no longer possible since a) 'launch' will trip over the multiweave bridge creation, and b) 'stop' will remove the bridge even though there are still weaves connected to it.

Suggested fix:
- only create multiweave bridge if it doesn't exist already
- remove ip addresses from multiweave when stopping individual weaves, and only delete multweave if there are no more addresses left

**Comment by rade** [comment]
> The old multiweave lets one launch additional peers, and also stop just some of the peers, by setting `FIRST_WEAVE`

...and now so does the new multiweave.

**Comment by rade** [comment]
For posterity, here is a patch to multiweave that makes it use just distinct ports instead of IPs. This does work but results in some "connection shutting down due to error: cannot connect to ourself" noise.

```patch
diff --git a/bin/multiweave b/bin/multiweave
index 40654b4..090776e 100755
--- a/bin/multiweave
+++ b/bin/multiweave
@@ -15,8 +15,6 @@ START=${FIRST_WEAVE:-1}
 COUNT=${NUM_WEAVES:-28}
 FINISH=$((START+COUNT-1))
 
-SUBNET=192.168.7
-
 weavedir=$(dirname $0)/..
 
 # Generate a random MAC value - copied from 'weave' script
@@ -30,24 +28,21 @@ random_mac() {
 weave() {
     i=$1
     shift
-    WEAVE_CONTAINER_NAME=weave$i WEAVE_HTTP_ADDR=$SUBNET.$i:6784 $weavedir/weave "$@"
+    WEAVE_CONTAINER_NAME=weave$i WEAVE_HTTP_ADDR=127.0.0.1:$((6000+i)) WEAVE_PORT=$((5000+i)) $weavedir/weave "$@"
 }
 
 case "$1" in
     launch)
         shift 1
         echo $(date -R) launching
-        ip link show multiweave >/dev/null 2>&1 || sudo ip link add name multiweave type bridge
         for i in $(seq $START $FINISH); do
-            IP=$SUBNET.$i
-            sudo ip addr add $IP/24 dev multiweave
-            weave $i launch-router --datapath '' --iface '' --no-dns --host $IP \
+            weave $i launch-router --datapath '' --iface '' --no-dns --port $((5000+i)) \
                 --name $(random_mac) --nickname weave$i "$@"
         done
         echo $(date -R) connecting
         [ $START -gt 1 ] || START=2
         for i in $(seq $START $FINISH); do
-            TO=$SUBNET.$((i-1))
+            TO=localhost:$((5000+i-1))
             echo connecting weave$i to $TO
             weave $i connect $TO
         done
@@ -55,12 +50,9 @@ case "$1" in
         ;;
     stop)
         for i in $(seq $FINISH -1 $START); do
-            IP=$SUBNET.$i
             echo Stopping weave$i
             weave $i stop-router
-            sudo ip addr del $IP/24 dev multiweave || true
         done
-        [ -z "$(ip -4 -o addr show multiweave)" ] && sudo ip link del multiweave
         ;;
     *)
             weave "$@"
```

**Pull request merged**